pkg/provisioning/provisioning.go

```diff
@@ -0,0 +1,98 @@
+package provisioning
+
+import (
+	metalkubev1alpha1 "github.com/metalkube/baremetal-operator/pkg/apis/metalkube/v1alpha1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("provisioner")
+
+/*
+Package provisioning defines the API for talking to the provisioning backend.
+*/
+
+// Provisioner holds the state information for talking to the
+// provisioning backend.
+//
+// NOTE(dhellmann): Provisioner will eventually become an interface,
```

Comment:
As noted on the other PR, I think we'll want the interface as soon as we write any tests.